text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,153 @@
+# Framework for defining the maturity of OpenTelemetry and its SIGs
+
+On 08 Mar 2023, the OpenTelemetry GC and TC held an OpenTelemetry Leadership summit, discussing various topics. One of the themes we discussed was establishing standard rules for describing the maturity of the OpenTelemetry project. This OTEP summarizes what was discussed there and is intended to have the wider community provide feedback.
+
+This OTEP builds on what was previously communicated by the project, especially the [Versioning and stability for OpenTelemetry clients](https://opentelemetry.io/docs/reference/specification/versioning-and-stability).
+
+The Collector's [stability levels](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) inspired the maturity levels.
+
+## Motivation
+
+Quite often, the community is faced with the question of the quality and maturity expectations of its diverse set of components. This OTEP aims to bring clarity by establishing a framework to communicate the project's maturity. As the OpenTelemetry project comprises a multitude of SIGs, and each SIG has several components of varying quality, having this framework will help set the right expectations for OpenTelemetry users. The public message we want to convey by stating that a SIG is a tier-1 SIG is that the SIG has both a healthy community and is seen as critical to the project as a whole.
```

Comment:
> the community is faced with the question of the quality and maturity expectations of its diverse **set of components**

(emphasis mine) +1 that this is a problem. But then you're making a leap from this to the question of "SIG maturity", which is a different problem altogether. Why is SIG maturity a problem?

Reply:
"SIG maturity" is quite judgemental, and I think it's affecting what is really meant in the OTEP. Perhaps we don't need to have SIGs being "stable", but having SIGs report their tier-1 components and their stability levels? OTel would then be ready to be called stable when all components for all tier-1 SIGs are marked as stable. Would it help to remove this intermediary level?

Reply:
SiG != github repository or contributor ecosystem necessarily.  IF we were to have any "bar" around stability and participation, I think it needs to be around things like release-velocity, major bug resolution time and contributor-health vs. attending the public meetings.